Matter_Engine/Query/VolumetricMaterialTakeoff.cs

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+using BH.oM.Physical.Materials;
+using BH.oM.Quantities.Attributes;
+using BH.oM.Base.Attributes;
+using BH.Engine.Base;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace BH.Engine.Matter
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Gets the unique Materials along with their volumes defining an object's make-up.")]
+        [Input("elementM", "The element to get the VolumetricMaterialTakeoff from.")]
+        [Input("checkForTakeoffFragment", "If true and the provided element is a BHoMObject, the incoming item is checked if it has a VolumetricMaterialTakeoff fragment attached, and if so, returns it. If false, the VolumetricMaterialTakeoff returned will be calculated, independant of fragment attached.")]
+        [Output("volumetricMaterialTakeoff", "The kind of matter the element is composed of and in which volumes.")]
+        public static VolumetricMaterialTakeoff IVolumetricMaterialTakeoff(this IElementM elementM, bool checkForTakeoffFragment = false)
```

Comment:
`checkForTakeoffFragment` should be defaulted to `true` IMO, otherwise the UX of the Revit workflows will be horrible.

Reply:
After discussion with @pawelbaran , will change to true, and add a Note if a fragment is found